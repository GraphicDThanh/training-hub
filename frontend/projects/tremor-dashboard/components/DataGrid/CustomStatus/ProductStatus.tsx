// Components
import { Button } from "@/components";

// Icons
import { MdDone, MdReplay, MdClose } from "react-icons/md";

// Constants
import { VARIANT_BUTTON } from "@/constants";

interface ProductStatusProps {
  status: number;
}

const ProductStatus = ({ status }: ProductStatusProps) => {
  switch (status) {
    case 0:
      return (
        <Button
          aria-label="Done"
          additionalClass="border-green-500 cursor-default"
          variant={VARIANT_BUTTON.LIGHT_STATUS}
          variantTremor={VARIANT_BUTTON.LIGHT}>
          <MdDone className="text-xs text-green-500" />
        </Button>
      );
    case 1:
      return (
        <Button
          aria-label="Close"
          additionalClass="border-red-500 cursor-default"
          variant={VARIANT_BUTTON.LIGHT_STATUS}
          variantTremor={VARIANT_BUTTON.LIGHT}>
          <MdClose className="text-xs text-red-500" />
        </Button>
      );
    case 2:
      return (
        <Button
          aria-label="Replay"
          additionalClass="border-primary dark:border-primary cursor-default"
          variant={VARIANT_BUTTON.LIGHT_STATUS}
          variantTremor={VARIANT_BUTTON.LIGHT}>
          <MdReplay className="text-xs text-primary dark:text-primary" />
        </Button>
      );
    default:
      return <span className="w-[25px] h-[25px] mr-2"></span>;
  }
};

export default ProductStatus;
