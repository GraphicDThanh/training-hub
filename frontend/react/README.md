# React Redux Training
[Plan doc](https://docs.google.com/document/d/1lYywN0RIhrwsyFZ8uPZuaN5QM78mYuXFD608vS8YEbM/edit?pli=1#)
## ES6
## React Basic
### Practice 1 - examples React
## React Advanced
### Practice 2
## Redux Basic
## Redux Advanced
## Redux Recipes
