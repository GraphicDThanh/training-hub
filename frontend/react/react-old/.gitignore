node_modules
out
dist
.DS_Store
.git
.docpad.db
vendor/
.vagrant
bower-components/
bower_components/
npm-debug.log
.gitignore~
vagrant/Vagrantfile~
build/
