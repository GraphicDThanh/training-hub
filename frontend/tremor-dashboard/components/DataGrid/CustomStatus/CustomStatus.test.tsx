import { render } from "@testing-library/react";

// Components
import CustomStatus from "./CustomStatus";

describe("CustomStatus component", () => {
  it("should render snapshot correctly", () => {
    const { container } = render(<CustomStatus status={0} />);

    expect(container).toMatchSnapshot();
  });

  it("should render Paid content if status is 0", () => {
    const { getByText } = render(<CustomStatus status={0} />);

    expect(getByText("Paid")).toBeTruthy();
  });

  it("should render Canceled content if status is 1", () => {
    const { getByText } = render(<CustomStatus status={1} />);

    expect(getByText("Canceled")).toBeTruthy();
  });

  it("should render Refunded content if status is 2", () => {
    const { getByText } = render(<CustomStatus status={2} />);

    expect(getByText("Refunded")).toBeTruthy();
  });
});
