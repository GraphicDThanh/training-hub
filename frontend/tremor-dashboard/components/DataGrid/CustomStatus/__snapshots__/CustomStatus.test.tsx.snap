// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomStatus component should render snapshot correctly 1`] = `
<div>
  <div
    class="flex justify-start items-center text-xs dark:text-lighter font-semibold leading-[15px] tracking-[0.4px] capitalize order-status w-[120px]"
  >
    <button
      aria-label="Done"
      class="tremor-Button-root shrink-0 inline-flex group font-medium outline-none text-sm dark:text-dark-tremor-brand bg-transparent dark:hover:text-dark-tremor-brand-emphasis w-[25px] h-[25px] justify-center items-center rounded-full border mr-2 text-tremor-content-title hover:text-tremor-content-title border-green-500 cursor-default"
    >
      <span
        class="tremor-Button-text text-tremor-default whitespace-nowrap"
      >
        <svg
          class="text-xs text-green-500"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
          />
        </svg>
      </span>
    </button>
    <p
      class="text-tremor-default text-xs text-tertiary dark:text-lighter font-semibold leading-[15px] tracking-[0.4px] capitalize order-status"
    >
      Paid
    </p>
  </div>
</div>
`;
